add-chain/main.go

```diff
@@ -176,12 +188,28 @@ func entrypoint(ctx *cli.Context) error {
 		return fmt.Errorf("failed to retrieve L1 rpc url: %w", err)
 	}
 
-	err = readAddressesFromChain(contractAddresses, l1RpcUrl)
+	// Portal version `3` is the first version of the `OptimismPortal` that supported the fault proof system.
+	version, err := castCall(addresses[OptimismPortalProxy], "version()(string)", l1RpcUrl)
+	if err != nil {
+		return fmt.Errorf("failed to get OptimismPortalProxy.version(): %w", err)
+	}
+
+	version, err = strconv.Unquote(version)
+	if err != nil {
+		return fmt.Errorf("failed to parse OptimismPortalProxy.version(): %w", err)
+	}
+	majorVersion, err := strconv.ParseInt(strings.Split(version, ".")[0], 10, 32)
+	if err != nil {
+		return fmt.Errorf("failed to parse OptimismPortalProxy.version(): %w", err)
+	}
+	isFPAC := majorVersion >= 3
```

Comment:
I suspect we want to just call this "isFaultProof", so I filed this https://github.com/ethereum-optimism/superchain-registry/issues/301 .